examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -0,0 +1,301 @@
+        0x7c,
+    };
+
+    return CopySpanToMutableSpan(ByteSpan{ kCdForAllExamples }, out_cd_buffer);
+
+#else
+    size_t buflen = out_cd_buffer.size();
+    ChipLogDetail(Crypto, "Get certificate declaration from se05x");
+    ReturnErrorOnFailure(se05xGetCertificate(CD_CERT_DECLARATION_DATA_SE05X_ID, out_cd_buffer.data(), &buflen));
+    out_cd_buffer.reduce_size(buflen);
+    return CHIP_NO_ERROR;
+#endif
+}
+
+CHIP_ERROR ExampleSe05xDACProviderv2::GetFirmwareInformation(MutableByteSpan & out_firmware_info_buffer)
+{
+    // TODO: We need a real example FirmwareInformation to be populated.
+    out_firmware_info_buffer.reduce_size(0);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ExampleSe05xDACProviderv2::SignWithDeviceAttestationKey(const ByteSpan & message_to_sign,
+                                                                   MutableByteSpan & out_signature_buffer)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    VerifyOrReturnError(IsSpanUsable(out_signature_buffer), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(IsSpanUsable(message_to_sign), CHIP_ERROR_INVALID_ARGUMENT);
+
+    ChipLogDetail(Crypto, "Sign using DA key from se05x (Using internal sign)");
+
+    MutableByteSpan tlv_buffer = ((MutableByteSpan &) message_to_sign).SubSpan(1); // Exclude the Start container
+
+    TLV::TLVReader msg_reader;
+    TLV::Tag tagValue;
+
+    msg_reader.Init(tlv_buffer);
+    msg_reader.Next();
+
+    /* To be removed. Use common key id to sign message */
+    static bool sign_cert_decl_attest = 1;
+
+    if (sign_cert_decl_attest)
+    {
+        /* Skip certificate declaration tag */
+        msg_reader.Next();
+
+        tagValue = msg_reader.GetTag();
+        VerifyOrReturnError(TLV::ContextTag(2) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
+        /* Get attestation nonce */
+        ByteSpan attest_nonce;
+        msg_reader.Get(attest_nonce);
+        /* Set attestation nonce */
+        VerifyOrReturnError(CHIP_NO_ERROR ==
+                                se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
+                            CHIP_ERROR_INTERNAL);
+
+        msg_reader.Next();
+        tagValue = msg_reader.GetTag();
+        VerifyOrReturnError(TLV::ContextTag(3) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
+        ByteSpan time_stamp;
+        msg_reader.Get(time_stamp);
+        uint8_t tmp = time_stamp.size();
+        /* Set time stamp length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tmp, 1), CHIP_ERROR_INTERNAL);
+        if (time_stamp.size() > 0)
+        {
+            /* Set time stamp data */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_TIME_STAMP_DATA_SE05X_ID, time_stamp.data(), time_stamp.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+
+        if (message_to_sign.size() >= 16)
```

Comment:
This doesn't really guarantee that we have not already read to within <16 bytes of the end, right?  Shouldn't this be comparing `message_to_sign.size()` minus the number of bytes the reader has consumed, at least?

Reply:
updated the condition now